include/mega/drivenotify.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file mega/drivenotify.h
+ * @brief Mega SDK various utilities and helper classes
+ *
+ * (c) 2013-2020 by Mega Limited, Auckland, New Zealand
+ *
+ * This file is part of the MEGA SDK - Client Access Engine.
+ *
+ * Applications using the MEGA API must present a valid application key
+ * and comply with the the rules set forth in the Terms of Service.
+ *
+ * The MEGA SDK is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * @copyright Simplified (2-clause) BSD License.
+ *
+ * You should have received a copy of the license along with this
+ * program.
+ */
+
+#pragma once
+
+#ifdef USE_DRIVE_NOTIFICATIONS
+
+#include <functional>
+#include <string>
+#include <queue>
+#include <mutex>
+
+namespace mega {
+
+
+    // Structure containing relevant Drive info.
+    // Windows: information is provided by Windows Management Instrumentation (WMI), Microsoft's implementation of WBEM.
+    struct DriveInfo                          //    Local               Removable/USB          Network
+    {                                         //
+        std::wstring mountPoint;              // C:                   E:                    F:
+        std::wstring location;                // ""/null              ""/null               \\host\f
+        std::wstring volumeSerialNumber;      // EE82D138             0EEE1DE2              A01A541C
+
+        // probably less useful
+        std::wstring size;                    // 1005343207424        31020957696           843572047872
+        std::wstring description;             // Local Fixed Disk     Removable Disk        Network Connection
+        uint32_t     driveType = 0;           // 3    (Fixed)         2      (Removable)    4      (Network)
+        uint32_t     mediaType = 0;           // 12   (Fixed HD)      0/null (Unknown)      0/null (Unknown)
+
+        bool         connected = false;
+    };
+
+
+
+    // Interface for receiving drive [dis]connection events, and notifying futher.
+    //
+    // Platform specific implementations:
+    // - DriveNotifyWin;
+    // - DriveNotifyPosix.
+    class DriveNotify
+    {
+    public:
+        bool start(std::function<void()> notify);
+        void stop();
+
+        std::pair<std::wstring, bool> get();
```

Comment:
This getter may need to be declared within an `#ifdef WIN32`:
```suggestion cpp
#ifdef WIN32
    std::pair<std::wstring, bool> get();
#elseif
    std::pair<std::string, bool> get();
#endif
```
Otherwise, on non-windows, the call to the callback in the `exec()` ([here](https://github.com/meganz/sdk/pull/2441/files#diff-fe6b11e559efca285b6c1c20b3de9a4cd3aba4c65837f872665ec883fdfa03b4R3141)) will throw a compiler error:
```
/home/sergio/dev/sdk/src/megaclient.cpp:3141: error: no matching function for call to ‘mega::LocalPath::fromPlatformEncoded(std::remove_reference<std::__cxx11::basic_string<wchar_t>&>::type)’
         app->drive_presence_changed(di.second, LocalPath::fromPlatformEncoded(move(di.first)));
                                                                                             ^
```
(Note: there `fromPlatformEncoded()` is overloaded for WIN32 in order to accept wide characters strings)

Reply:
Fixed in a4f24c1